File: tests/python/windowing.py
# import time
import numpy as np

import brainflow
from brainflow.board_shim import BoardShim, BrainFlowInputParams, LogLevels, BoardIds
from brainflow.data_filter import DataFilter, FilterTypes, WindowFunctions


def main():
    # demo for data windowing
    window_len = 20

    for window_function in range(4):

        if window_function == WindowFunctions.NO_WINDOW.value:
            print('Window data for NO_WINDOW function:')
            window_data = DataFilter.get_window(WindowFunctions.NO_WINDOW.value, window_len)
        elif window_function == WindowFunctions.HANNING.value:
            print('Window data for HANNING function:')
            window_data = DataFilter.get_window(WindowFunctions.HANNING.value, window_len)
        elif window_function == WindowFunctions.HAMMING.value:
            print('Window data for HAMMING function:')
            window_data = DataFilter.get_window(WindowFunctions.HAMMING.value, window_len)
        else:
            print('Window data for BLACKMAN_HARRIS function:')
            window_data = DataFilter.get_window(WindowFunctions.BLACKMAN_HARRIS.value, window_len)

        print(window_data)


if __name__ == "__main__":
    main()

File: java-package/brainflow/src/main/java/brainflow/examples/Windowing.java
package brainflow.examples;

import java.util.Arrays;

import brainflow.DataFilter;

public class Windowing
{

    public static void main (String[] args) throws Exception
    {
        int window_len = 20;
        for (int window_function = 0; window_function < 4; window_function++)
        {
            double[] window_data = DataFilter.get_window (window_function, window_len);
            System.out.println ("Window data:");
            System.out.println (Arrays.toString (window_data));
        }
    }
}
